extensions-core/lookups-cached-global/src/main/java/io/druid/server/lookup/namespace/cache/OffHeapNamespaceExtractionCacheManager.java

```diff
@@ -48,7 +48,7 @@
   private static final Logger log = new Logger(OffHeapNamespaceExtractionCacheManager.class);
   private final DB mmapDB;
   private ConcurrentMap<String, String> currentNamespaceCache = new ConcurrentHashMap<>();
-  private Striped<Lock> nsLocks = Striped.lock(32); // Needed to make sure delete() doesn't do weird things
+  private Striped<Lock> nsLocks = Striped.lazyWeakLock(1024); // Needed to make sure delete() doesn't do weird things
```

Comment:
quite simply:  Increased the quantity of locks to reduce the likelihood of hash collision. Lazy Weak to prevent needing to have 1024 in memory all the time.

LazyWeak refers to the REFERNCE to ReentrantLock, *NOT* the lock itself. i.e. a `ReentrantLock` is generated lazily and kept as a weak reference, that way if noone is actively locking it, it is candidate for garbage collection. But if someone is holding a strong reference (aka, holds a lock) then it will not be garbage collected.